apps/drec-ui/src/apps/device-group/logic/smartMeter/chart/useGenerateChartLabels.ts

```diff
@@ -14,14 +16,18 @@ export const useGenerateChartLabels = ({
     format
 }: TUseGenerateChartLabelsArgs) => {
     const labels: string[] = [];
-    let current = start;
 
-    while (dayjs(current).isBefore(dayjs(end))) {
+    for (
+        let current = start;
+        current < end;
+        current = dayjs(current).add(multiplier, TimeUnitPluralEnum.minutes).toDate()
+    ) {
+        if (labels.includes(dayjs(current).format(format))) {
```

Comment:
way too hacky. needs to be improved

Reply:
I know but this issue only happens for Week and Years. For days and months it is working fine. That`s why it is a bit weird. I will try to improve it.

Reply:
Found the issue. It was the multiplier minutes. Should work fine now.